packages/purpose-process/test/testArchivePurposeVersion.ts

```diff
@@ -0,0 +1,239 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  getMockPurposeVersion,
+  getMockPurpose,
+  readLastEventByStreamId,
+  decodeProtobufPayload,
+} from "pagopa-interop-commons-test";
+import {
+  PurposeVersion,
+  purposeVersionState,
+  Purpose,
+  generateId,
+  PurposeArchivedV2,
+  toPurposeV2,
+  PurposeId,
+  PurposeVersionId,
+  TenantId,
+} from "pagopa-interop-models";
+import { describe, expect, it, vi } from "vitest";
+import {
+  purposeNotFound,
+  organizationIsNotTheConsumer,
+  purposeVersionNotFound,
+  notValidVersionState,
+} from "../src/model/domain/errors.js";
+import {
+  postgresDB,
+  purposes,
+  purposeService,
+} from "./purposeService.integration.test.js";
+import { addOnePurpose } from "./utils.js";
+
+export const testArchivePurposeVersion = (): ReturnType<typeof describe> =>
+  describe("archivePurposeVersion", () => {
+    it("should write on event-store for the archiving of a purpose version", async () => {
+      vi.useFakeTimers();
+      vi.setSystemTime(new Date());
+
+      const mockPurposeVersion: PurposeVersion = {
+        ...getMockPurposeVersion(),
+        state: purposeVersionState.active,
+      };
+      const mockPurpose: Purpose = {
+        ...getMockPurpose(),
+        versions: [mockPurposeVersion],
+      };
+      await addOnePurpose(mockPurpose, postgresDB, purposes);
+
+      await purposeService.archivePurposeVersion({
+        purposeId: mockPurpose.id,
+        versionId: mockPurposeVersion.id,
+        organizationId: mockPurpose.consumerId,
+        correlationId: generateId(),
+      });
+
+      const writtenEvent = await readLastEventByStreamId(
+        mockPurpose.id,
+        "purpose",
+        postgresDB
+      );
+
+      expect(writtenEvent).toMatchObject({
+        stream_id: mockPurpose.id,
+        version: "1",
+        type: "PurposeArchived",
+        event_version: 2,
+      });
+
+      const expectedPurpose: Purpose = {
+        ...mockPurpose,
+        versions: [
+          {
+            ...mockPurposeVersion,
+            state: purposeVersionState.rejected,
+            updatedAt: new Date(),
```

Comment:
This is so much cleaner than in other tests where we get the date from the event, we should update those and doing like this